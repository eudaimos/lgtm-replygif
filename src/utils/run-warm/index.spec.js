import runWarm from '.';

describe('[Utility: runWarm]', () => {
  it('should return early if the event is from the scheduler', async () => {
    const handler = runWarm(() => {});

    expect(await handler({
      source: 'aws.events',
    })).toMatchSnapshot();
  });

  it('should invoke the handler function', async () => {
    const handler = jest.fn();
    const warmHandler = runWarm(handler);

    await warmHandler({});
    expect(handler).toHaveBeenCalled();
  });
});
