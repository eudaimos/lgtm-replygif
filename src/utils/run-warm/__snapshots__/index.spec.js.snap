// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Utility: runWarm] should return early if the event is from the scheduler 1`] = `"pinged"`;
